### PR TITLE
feat(ci): upload logs of failed builds

### PR DESCRIPTION
**Pull request by dpc**
Example: https://github.com/fedimint/fedimint/actions/runs/5895608638?pr=2980

image

### CODE REVIEW

**Comment by elsirion** [comment]
Failed to fail properly :frowning:

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -102,11 +102,32 @@ jobs:
...
+        run: |
+          nix build -L .#ci.cli-test.all --keep-failed
 
       - name: Wasm Tests
         if: (github.ref == 'refs/heads/master' || matrix.build-in-pr) && matrix.run-tests
-        run: nix build -L .#ci.wasm-tests
+        run: nix build -L .#ci.wasm-tests --keep-failed
+
+      - name: Prepare failed test build dirs
+        if: always()
```
Comment (douglaz) [inline .github/workflows/ci-nix.yml:113]: Does github really need this if?

**Review by douglaz — COMMENTED** [review]
LGTM

File: .github/workflows/ci-nix.yml (hunk 1 above)
Comment (dpc) [reply .github/workflows/ci-nix.yml:113]: Otherwise if previous steps fail (tests), these actions won't execute, no?

**Review by justinmoon — APPROVED** [review]

File: .github/workflows/ci-nix.yml (hunk 1 above)
Comment (douglaz) [reply .github/workflows/ci-nix.yml:113]: :thinking: 
Not sure, but whatever

**Pull request merged**